plugin/environment.go

```diff
@@ -95,14 +97,34 @@ func (env *Environment) Active() []*model.BundleInfo {
 
 // IsActive returns true if the plugin with the given id is active.
 func (env *Environment) IsActive(id string) bool {
-	_, ok := env.activePlugins.Load(id)
-	return ok
+	return env.GetPluginState(id) == model.PluginStateRunning
+}
+
+// GetPluginState returns the current state of a plugin (disabled, running, or error)
+func (env *Environment) GetPluginState(id string) int {
+	rp, ok := env.registeredPlugins.Load(id)
+	if !ok {
+		return model.PluginStateNotRunning
+	}
+
+	return *rp.(*registeredPlugin).State
+}
+
+// SetPluginState sets the current state of a plugin (disabled, running, or error)
+func (env *Environment) SetPluginState(id string, state int) bool {
```

Comment:
Maybe (1/5) change the return value to `error`? Trying to set the state that is not running, seams to be an error case to me.